mne/epochs.py

```diff
@@ -1342,6 +1342,8 @@ def _get_data(self, out=True, verbose=None):
                         epoch = self._data[idx]
                 else:  # from disk
                     epoch_noproj = self._get_epoch_from_raw(idx)
+                    if epoch_noproj is None:  # Dropped due to bad segment
+                        continue
```

Comment:
Why do you need this? The following two functions should already have this short-circuit builtin IIRC so this is less DRY